docs/walletkit/android/chain-abstraction.mdx

```diff
@@ -0,0 +1,233 @@
+
+For example, when an app requests a 100 USDC payment on Base network but the user only has USDC on Arbitrum, WalletKit offers methods to detect this mismatch, generate necessary transactions, track the cross-chain transfer, and complete the original transaction after bridging finishes.
+
+:::note
+Chain abstraction is currently in experimental phase and requires the `@ChainAbstractionExperimentalApi` annotation.
+:::
+
+## How It Works
+
+When a dapp requests a transaction through `eth_sendTransaction`, you need to:
+1. Detect if chain abstraction is needed
+2. Generate necessary bridging transactions
+3. Handle transaction execution and monitoring
+4. Complete the original transaction once bridging is done
+
+The following sequence diagram illustrates the complete flow of a chain abstraction operation, from the initial dapp request to the final transaction confirmation
+
+![Chain Abstraction Flow](/assets/chain-abstraction-flow.png)
+
+## Methods
+
+Following are the methods from WalletKit that you will use in implementing chain abstraction.
+
+### Prepare 
+
+This method is used to check if chain abstraction is needed. If it is, it will return a `FulfilmentSuccess.Available` object with the necessary transactions and funding information. 
+If it is not, it will return a `FulfilmentSuccess.NotRequired` object with the original transaction. 
+
+```kotlin 
+@ChainAbstractionExperimentalApi
+fun prepare(
+  transaction: Wallet.Model.Transaction,
+  onSuccess: (Wallet.Model.FulfilmentSuccess) -> Unit,
+  onError: (Wallet.Model.FulfilmentError) -> Unit
+)
+```
+
+### Status 
+
+This method is used to check the status of the fulfillment operation. It will return a `FulfilmentStatus.Completed` object if the operation completed successfully or a `FulfilmentStatus.Error` object if it encountered an error.
+
+```kotlin
+@ChainAbstractionExperimentalApi
+fun status(
+  fulfilmentId: String,
+  checkIn: Long,
+  onSuccess: (Wallet.Model.FulfilmentStatus.Completed) -> Unit,
+  onError: (Wallet.Model.FulfilmentStatus.Error) -> Unit
+)
+```
+
+### Estimate Fees
+
+This method is used to estimate the fees for the fulfillment operation.
+
+```kotlin
+@Throws(Exception::class)
+@ChainAbstractionExperimentalApi
+fun estimateFees(chainId: String): Wallet.Model.EstimatedFees
+```
+
+### Get ERC20 Balance
+
+This method is used to get the balance of an ERC20 token on a specific chain.
+
+```kotlin
+@Throws(Exception::class)
+@ChainAbstractionExperimentalApi
+fun getERC20Balance(chainId: String, tokenAddress: String, ownerAddress: String): String 
+```
+
+### Get Fulfilment Details
+
+This method is used to get the details of the fulfillment operation.
+
+```kotlin
+@ChainAbstractionExperimentalApi
+fun getFulfilmentDetails(
```

Comment:
Most likely I'll remove the fulfilment wording from the API but I'll commit changes directly to the PR